### ISSUE TITLE
Possible label leakage during inference?

### ISSUE THREAD

**Issue by X-funbean** [mainaksingha01/ODG-CLIP#5]
According to the paper,  the inference should should involve generating latent representations with respect to all the $$C+1$$ class labels
<img width="287" alt="image" src="https://github.com/user-attachments/assets/7f773689-57ff-4042-9c45-ff44923a855e">
However, in the code implementation, the `CustomCLIP` used for inference shares the same `forward` function for training, where the latent representations are generated according to the ground-truth class labels of the samples. This may result in label leakage during inference. May I ask whether my understanding on this issue is correct?

https://github.com/mainaksingha01/ODG-CLIP/blob/master/trainer/odgclip.py#L375-L400
```python3
values_tensor = []
for i, value in zip(range(len(norm_textfeatures)), label):
    classtext_value = norm_textfeatures[i, value, :]
    values_tensor.append(classtext_value)
matched_texts = torch.stack(values_tensor) 

self.upsample_net = self.upsample_net.to(device)
upsampled_texts = self.upsample_net(matched_texts)
target_size = (224, 224)
final_texts = F.interpolate(upsampled_texts, size=target_size, mode='bilinear', align_corners=False)

new_image = torch.cat((image, final_texts), dim=1).to(device)
final_image = self.conv_layer(new_image.to(torch.float32))


mean = [0.48145466, 0.4578275, 0.40821073]
std = [0.26862954, 0.26130258, 0.27577711]

normalize = transforms.Normalize(mean=mean, std=std)
rescaled_image = normalize(final_image)
image_features, _ = self.image_encoder(rescaled_image.type(self.dtype))
image_features = image_features / image_features.norm(dim=-1, keepdim=True)
```
